### ISSUE TITLE
Refine Creation of Data Split

### ISSUE THREAD

**Issue by stdhd** [stdhd/inverse-sketcher#7]
There are generally two possible cases when thinking of creating a test split, these need to be implemented seperately.

- [ ] Splitting data along the labels, so labels in test set have not seen before in training data

- [ ] Setting aside a test fraction of images and corresponding sketches for each label
